spec/abci/abci++_app_requirements.md

```diff
@@ -188,6 +192,39 @@ Likewise, `ExtendVote` can also be non-deterministic:
 * *w<sup>r</sup><sub>p</sub> = w<sup>r</sup><sub>q</sub> &#8655;
   e<sup>r</sup><sub>p</sub> = e<sup>r</sup><sub>q</sub>*
 
+### Mempool Connection Requirements
+
+Let *CheckTxCodes<sub>tx,p,h</sub>* denote the set of result codes returned by *p*'s Application,
+via `ResponseCheckTx`,
+to successive calls to `RequestCheckTx` occurring while the Application is at height *h*
+and having transaction *tx* as parameter.
+*CheckTxCodes<sub>tx,p,h</sub>* is a set since *p*'s Application may
+return different result codes during height *h*.
+If *CheckTxCodes<sub>tx,p,h</sub>* is a singleton set, i.e. the Application always returned
+the same result code in `ResponseCheckTx` while at height *h*,
+we define *CheckTxCode<sub>tx,p,h</sub>* as the singleton value of *CheckTxCodes<sub>tx,p,h</sub>*.
+If *CheckTxCodes<sub>tx,p,h</sub>* is not a singleton set, *CheckTxCode<sub>tx,p,h</sub>* is undefined.
+Let predicate *OK(CheckTxCode<sub>tx,p,h</sub>)* denote whether *CheckTxCode<sub>tx,p,h</sub>* is `SUCCESS`.
+
+* Requirement 13 [`CheckTx`, eventual non-oscillation]: There exists a boolean value *b*,
+  and a height *h<sub>stable</sub>*  such that
+  for any correct process *p*
+  and any transaction *tx*,
```

Comment:
I think the order is not what is intended here. From the requirement as written here, all `tx` must have the same `b`. From the English text below it seems that you want: 
for all `tx`, exists `b`, exists `h`, forall `p`
This also would mean that from some height on, all processes agree on the outcome. This should also be highlighted.

Reply:
Fixed. Please take a look

Reply:
Looks good. I still would add a comment on that all processes must agree on that stable value of `b`.

Reply:
@otrack Yes, I tried (1) first but, unlike in the notation you are using, in a notation consistent with the rest of the ABCI spec (all requirements) above I found it very hard to understand... so in the end I went for (2), and explain that, in practice, App devs can just enforce (1)... since as you say (1) implies (2).

We could adapt the whole spec to use a more advanced notation as the one you are using, but I'm not sure now's the moment to do that.

Reply:
> Looks good. I still would add a comment on that all processes must agree on that stable value of b.

I added sentence at the end

Reply:
Thanks! The new sentence looks good. I think it makes sense that you didn't use the term agree, for the reasons @otrack mentioned above.